src/widget/wfindonmenu.cpp

```diff
@@ -0,0 +1,279 @@
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::addActionsTrackTitle(Service service,
+        const QString& trackTitle,
+        QMenu* pServiceMenu,
+        TrackProperty trackProperty) {
+    const QString prefixAction = composePrefixAction(trackProperty);
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::openInBrowser(Service service,
+        TrackProperty trackProperty,
+        const QString& query) {
+    const QString serviceUrl = getServiceUrl(service, trackProperty);
+    QUrlQuery qurlquery;
+    qurlquery.addQueryItem("q", query);
+    QUrl url(serviceUrl);
+    url.setQuery(qurlquery);
+    QDesktopServices::openUrl(url);
+}
+
+void WFindOnMenu::populateWebLookUpQueries(QMenu* pServiceMenu,
+        const Track& track,
+        const QString& serviceTitle,
+        Service service) {
+    const auto artistTitle = track.getArtist();
```

Comment:
What's an "artist title"??

Reply:
I will change it as "primaryArtist".

Reply:
Why not just artist?

Sometimes I have artists like
"Foo feat. Bar"

I don't think we will be able to extract the primary artist.

Reply:
Just artist is okay :+1: .

I thought for later, we might want to add something with "album artist" as a second artist lookup. So if this become wanted addition, we can think about adding this primaryArtist and secondaryArtist as album artist.

Reply:
That route is not easy because the album artist ist often not even populate. It is interesting If it is a featuring Album like Apocalyptica, recording their track with different vocal artists.

Reply:
> It is interesting If it is a featuring Album like Apocalyptica, recording their track with different vocal artists.

Would be nice to have then, but should I implement it right now or the later?